### PR TITLE
CloudioEnpoint: modify constructor allowing uuid set from .properties

### PR DESCRIPTION
**Pull request by RandomReaper**
```new CloudioEnpoint("application_name")``` will look for ```application_name.properties``` in the usual places, then set the uuid from the ```ch.hevs.cloudio.endpoint.uuid``` property.

Using that constructor, the application does not need to know the uuid before loading the configuration file.

### CODE REVIEW

File: src/main/java/ch/hevs/cloudio/endpoint/CloudioEndpoint.java [hunk 1]
```diff
@@ -377,6 +383,7 @@ public void removeNode(String nodeName) {
     /*** Internal API *************************************************************************************************/
     class InternalEndpoint implements CloudioNodeContainer, MqttCallback, Runnable {
         /*** Constants ************************************************************************************************/
+        private static final String UUID			                = "ch.hevs.cloudio.endpoint.uuid";
```
Comment (cm0x4D) [inline src/main/java/ch/hevs/cloudio/endpoint/CloudioEndpoint.java]: `UUID_PROPERTY`

File: src/main/java/ch/hevs/cloudio/endpoint/CloudioEndpoint.java [hunk 2]
```diff
@@ -192,7 +197,8 @@
      * The endpoint will try immediately to connect to the central message broker and tries automatically to maintain
      * this connection in the background.
      *
-     * @param uuid                              Unique ID of the endpoint.
+     * @param uuid                              Unique ID of the endpoint or a filename if the {filename}.properties file
```
Comment (cm0x4D) [inline src/main/java/ch/hevs/cloudio/endpoint/CloudioEndpoint.java]: - Rename Argument to something like `uuidOrApplicationName`
- Update documentation for all constructors

File: src/main/java/ch/hevs/cloudio/endpoint/CloudioEndpoint.java [hunk 3]
```diff
@@ -429,13 +436,6 @@ public InternalEndpoint(String uuid, CloudioEndpointConfiguration configuration,
                 throw new InvalidUuidException("Uuid can not be null!");
             }
 
-            // Set the UUID.
-            this.uuid = uuid;
```
Comment (cm0x4D) [inline src/main/java/ch/hevs/cloudio/endpoint/CloudioEndpoint.java:433]: Rename the constructor's parameter to something like `uuidOrApplicationName'

**Review by cm0x4D — CHANGES_REQUESTED** [review]

**Comment by cm0x4D** [comment]
This will become really handy when the deployment service of *0.2* is ready where it is possible to download the complete configuration + all required certificates directly from the REST API.

File: src/main/java/ch/hevs/cloudio/endpoint/CloudioEndpoint.java (hunk 1 above)
Comment (cm0x4D) [reply src/main/java/ch/hevs/cloudio/endpoint/CloudioEndpoint.java]: @RandomReaper To be consistent, we should use UUID_PROPERTY for the key like all others

**Comment by RandomReaper** [comment]
take #3 contains garbage!

**Comment by cm0x4D** [comment]
> take #3 contains garbage!

I would rather say the start of a good idea ;-)

**Review by cm0x4D — APPROVED** [review]

**Comment by cm0x4D** [comment]
Thanks @RandomReaper

**Pull request merged**